diff --git a/drivers/led/ws2812.h b/drivers/led/ws2812.h
index 8013e5bd2d..cae8dacdaf 100644
--- a/drivers/led/ws2812.h
+++ b/drivers/led/ws2812.h
@@ -17,6 +17,10 @@
 
 #include "util.h"
 
+#if defined(RGB_DI_PIN) && !defined(WS2812_DI_PIN)
+#    define WS2812_DI_PIN RGB_DI_PIN
+#endif
+
 /*
  * The WS2812 datasheets define T1H 900ns, T0H 350ns, T1L 350ns, T0L 900ns. Hence, by default, these
  * are chosen to be conservative and avoid problems rather than for maximum throughput; in the code,
@@ -89,6 +93,10 @@ typedef struct PACKED ws2812_led_t {
 #endif
 } ws2812_led_t;
 
+#ifndef LED_TYPE
+#    define LED_TYPE ws2812_led_t
+#endif
+
 void ws2812_init(void);
 void ws2812_set_color(int index, uint8_t red, uint8_t green, uint8_t blue);
 void ws2812_set_color_all(uint8_t red, uint8_t green, uint8_t blue);
